### ISSUE TITLE
Why Unity Level up use one plugin for each platform instead of relying on the core? 

### ISSUE THREAD

**Issue by 00christian00** [soomla/unity3d-levelup#9]
I'm just starting to use soomla and I noticed that there is a lot of extra code that could be removed.
In the Unity Level Up there is an implementation for each platform plus one generic for the editor.
Since the core is already abstracted and have it's own version for each platform, why isn't the code just relying on the core instead of implementing again all the methods as plugins?
It's reinventing the wheel.
For example here is the generic version of _setFastestDurationMillis:

	protected virtual void _setFastestDurationMillis(Level level, long duration) {
#if UNITY_EDITOR
			string key = keyFastestDuration (level.ID);
			string val = duration.ToString ();
			PlayerPrefs.SetString (key, val);
#endif
		}
and then KeyValueStorage

virtual protected void _setValue(string key, string val) {
#if UNITY_EDITOR
			PlayerPrefs.SetString (key, val);
#endif
		}
Which is then implemented for each platform. Why doesn't it simply call _setFastestDurationMillis KeyValueStorage SetString?
A lot of code could be avoided, simplifying it and resulting in less bugs and smaller footprint.

**Comment by refaelos** [comment]
Did you also post this on http://answers.soom.la ? If not, can you post it there?

**Comment by 00christian00** [comment]
Yes I did.
http://answers.soom.la/t/why-unity-level-up-use-one-plugin-for-each-platform-instead-of-relying-on-the-core/3381

I am also working on several improvements in my fork:
https://github.com/00christian00/unity3d-levelup/commits/master
I already did remove the plugin, once it's fully tested I will merge it back.

Can new methods unique to Unity be accepted into the framework? Or they must be present in all platform to be integrated?
For example I just added an AddLevelScore that automatically create and name the score using the level name.
